raiden/storage/sqlite.py

```diff
@@ -361,5 +354,23 @@ def get_events(self, limit: int = None, offset: int = None):
         entries = self._query_events(limit, offset)
         return [self.serializer.deserialize(entry[0]) for entry in entries]
 
+    def get_snapshots(self, raw: bool = False) -> Generator[Tuple[int, Any], None, None]:
```

Comment:
btw, here is a proposal. I think we should ensure

- the upgrade is executed inside a transaction, otherwise there is a risk of partial updates
- the upgrade is done with a callback, without exposing the internal identifiers of the database

to provide the above through a sane interface, the `SQLiteStorage` class can have a method `upgrade`:

```python
  def upgrade(self):
     return SQLiteUpgrade(self)

class SQLiteUpgrade:
  def __init__(self, sqlite):
    self.sqlite = sqlite
    # open transaction
  def __del__(self):
    # close transaction
  def upgrade_snapshot(self, upgrade: Function[dict, dict]) -> None:
    for id_, snapshot in self.fetch_all_snapshots():
      self.overwrite_snapshot(id_, sqlite.serializer.serialize(upgrade(snapshot)))
```

The above is just a sketch, but the upgrade class would be responsible to ensure a transaction is open, and remove the implementation detail of the database identifier and serialization format from the upgrade.